src/hotspot/share/classfile/modules.cpp

```diff
@@ -628,17 +628,20 @@ void Modules::serialize(SerializeClosure* soc) {
 }
 
 void Modules::dump_native_access_flag() {
+  ResourceMark rm;
   const char* native_access_names = get_native_access_flags_as_sorted_string();
   if (native_access_names != nullptr) {
     _archived_native_access_flags = ArchiveBuilder::current()->ro_strdup(native_access_names);
   }
 }
 
 const char* Modules::get_native_access_flags_as_sorted_string() {
+  assert(Thread::current()->current_resource_mark() != nullptr, "Setup by caller");
```

Comment:
I think these asserts are redundant. st.as_string() will fail if there are no resource marks. We have plenty of functions that return a resurce-allocated object and it's uncommon to add such asserts.

Reply:
Agreed. We detect missing ResourceMarks so there is no need to check for present ones.

Reply:
I added the assertion to prevent installing `ResourceMark` here, which will result in returning invalid string. @dholmes-ora suggested additional comment, but I prefer assertion.

Reply:
That is not the way we generally do things. The need for a ResourceMark is part of the contract of the method hence we document it in a comment. If the caller fails to provide one then we get an assertion failure. We don't put assertions like this inside these methods to guard against putting a RM in the method itself!